src/v/security/authorizer.h

```diff
@@ -26,6 +26,119 @@
 
 namespace security {
 
+/**
+ * Holds authZ check metadata for audit processing
+ */
+struct auth_result {
+    // Flag indicating if user is authorized
+    bool authorized{false};
+    // Indicates if the authorization system is disabled
+    bool authorization_disabled{false};
+    // Indicates if the user is a superuser
+    bool is_superuser{false};
+    // Indicates if no ACL matches were found
+    bool empty_matches{false};
+
+    // If found, the resource pattern that was matched to provide authZ decision
+    std::optional<std::reference_wrapper<const resource_pattern>>
+      resource_pattern;
+    // If found, the ACL that was matched that provided the authZ decision
+    std::optional<acl_entry_set::const_reference> acl;
```

Comment:
I'd just return copies here. Across scheduling points or other mutating methods to the underlying containers these results would not be guaranteed to be alive.

Reply:
Fair point.  There are no scheduling points from the `request_context::authorized` call down to the authorizer.  However there will be after this structure is enqueued in the auditing system.

Reply:
depending on how big these types are, i think returning references/iterators makes sense. there are two cases

1. no audit logging, no scheduling points, no copying, fast
2. audit logging--we have to make copies anyway, but the caller gets to make that decision

Reply:
yeh maybe it's too much optimization right now